src/Illuminate/Contracts/Routing/UrlGenerator.php

```diff
@@ -76,4 +76,18 @@ public function action($action, $parameters = [], $absolute = true);
      * @return $this
      */
     public function setRootControllerNamespace($rootNamespace);
+
+    /**
+     * Get the request instance.
+     *
+     * @return \Illuminate\Http\Request
+     */
+    public function getRequest();
+
+    /**
+     * Get the full URL for the current request.
+     *
+     * @return string
+     */
+    public function full();
```

Comment:
Might be best to move this method up closer to the `current` method.

Reply:
I have moved the method up.